numpy/core/_internal.py

```diff
@@ -288,22 +288,32 @@ def _newnames(datatype, order):
     raise ValueError("unsupported order value: %s" % (order,))
 
 # Given an array with fields and a sequence of field names
-# construct a new array with just those fields copied over
-def _index_fields(ary, fields):
-    from .multiarray import empty, dtype, array
+# construct new array with just those fields
+def _index_fields(ary, names):
+    from .multiarray import dtype, array
     dt = ary.dtype
 
-    names = [name for name in fields if name in dt.names]
-    formats = [dt.fields[name][0] for name in fields if name in dt.names]
-    offsets = [dt.fields[name][1] for name in fields if name in dt.names]
+    if ary.dtype.fields == None:
```

Comment:
Hmm I could have sworn I added this to fix a bug. Maybe it was in an old version of this PR where _index_fields took care of more cases. In any case I'll remove it now.